### PR TITLE
Fix <emails> in notifications

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Given this ticket:
image

The emails surrounded between< and >will be missing from the notification content:
image

Encoding these emails a second time before sending the notification seems to fix the issue.

After fix:
image

This fix feel kind of hacky, if someone want to spend time to find a better solution he is welcome to do so :)

Some thoughts on this:
* If you check the sources (tinymce) from the client you can cleary see that only the \<emails\> are encoded: 
image
* However in the database both the html tags and the \<emails\> are encoded:
image
* Either the \<emails\> should be double encoded in the database or the others tags should not be encoded at all (isn't encoding html in the database a bad practice anyway ?)

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !21113

### CODE REVIEW

File: inc/notificationtargetticket.class.php [hunk 1]
```diff
@@ -165,6 +165,15 @@ function getDataForObject(CommonDBTM $item, array $options, $simple = false) {
       $data = parent::getDataForObject($item, $options, $simple);
       /*$data['##ticket.description##'] = Html::clean($data['##ticket.description##']);*/
 
+      // Double encode emails stored between '<' and '>' tags
+      // Common case of this is when the ticket was created from a forwarded email
+      // Without double encoding, these emails are interpreted as html and not
+      // rendered in the final mail
+      $regex = "/(&lt;[A-Za-z0-9.!#$%&'*+\-\/=?^_`{|}~]*?@[A-Za-z0-9.!#$%&'*+\-\/=?^_`{|}~]*?&gt;)/";
```
Comment (cedric-anne) [inline inc/notificationtargetticket.class.php:172]: We are not validation the email address, so maybe the regex could be simplier.

Comment (AdrienClairembault) [reply inc/notificationtargetticket.class.php]: Ok, simplified it to accept any chars expect ";" (= encoded tag end).

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Comment by AdrienClairembault** [comment]
I've moved back the PR to the draft status as the internal ticket doesn't appear to be resolved.

**Comment by AdrienClairembault** [comment]
The first fix was OK for html notifications but not for text notifications.
Latest commit should fix the issue with text.

**Review by trasher — DISMISSED** [review]

**Review by orthagh — APPROVED** [review]

File: inc/html.class.php [hunk 2]
```diff
@@ -57,6 +57,12 @@ class Html {
    static function clean($value, $striptags = true, $keep_bad = 2) {
       $value = Html::entity_decode_deep($value);
 
+      // Change <email@domain> to email@domain so it is not removed by htmLawed
+      $regex = "/(<[^>]+?@[^;]+?>)/";
+      $value = preg_replace_callback($regex, function($matches) {
+         return substr($matches[1], 1, (count($matches[1]) - 2));
```
Comment (cedric-anne) [inline inc/html.class.php:63]:
```suggestion
         return substr($matches[1], 1, (strlen($matches[1]) - 2));
```

**Review by trasher — APPROVED** [review]

**Pull request merged**